File: Python/bill.py
#Author:Clayton
#Rollno:31
#Title:Goa Legislative Assembly
#Startdate:15-07-24
#Modifieddate:22-07-24
#Description:to show oop implementation in goa legislative assembly

class BillStatus:
    PROPOSED = "PROPOSED"
    PENDING = "PENDING"
    APPROVED = "APPROVED"
    REJECTED = "REJECTED"

class Bill:
    def __init__(self, title, description, proposer, date_introduced, status):
        self.title = title
        self.description = description
        self.proposer = proposer
        self.date_introduced = date_introduced
        self.status = status

    def get_title(self):
        return self.title

    def get_description(self):
        return self.description

    def get_proposer(self):
        return self.proposer

    def get_date_introduced(self):
        return self.date_introduced

    def get_status(self):
        return self.status

    def set_status(self, status):
        self.status = status

    def display_info(self):
        print("Bill Title:", self.title)
        print("Description:", self.description)
        print("Proposer:", self.proposer)
        print("Date Introduced:", self.date_introduced)
        print("Status:", self.status)

File: Java/Bill.java
//Author:Clayton
//Rollno:31
//Title:Goa Legislative Assembly
//Startdate:15-07-24
//Modifieddate:22-07-24
//Description:to show oop implementation in goa legislative assembly


public class Bill {
    public enum BillStatus {
        PROPOSED,
        PENDING,
        APPROVED,
        REJECTED
    }

    private String title;
    private String description;
    private String proposer;
    private String dateIntroduced;
    private BillStatus status;

    public Bill(String title, String description, String proposer, String dateIntroduced, BillStatus status) {
        this.title = title;
        this.description = description;
        this.proposer = proposer;
        this.dateIntroduced = dateIntroduced;
        this.status = status;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getProposer() {
        return proposer;
    }

    public String getDateIntroduced() {
        return dateIntroduced;
    }

    public BillStatus getStatus() {
        return status;
    }

    public void setStatus(BillStatus status) {
        this.status = status;
    }

    public void displayInfo() {
        System.out.println("Bill Title: " + title);
        System.out.println("Description: " + description);
        System.out.println("Proposer: " + proposer);
        System.out.println("Date Introduced: " + dateIntroduced);
        System.out.println("Status: " + status);
    }
}